### ISSUE TITLE
Setting nixpkgs.config in a module doesn't seem to work anymore

### ISSUE THREAD

**Issue by chvp** [gytis-ivaskevicius/flake-utils-plus#4]
I have some infrastructure in my nixos config to allow only specific unfree packages (to avoid accidental unfree use). Therefore I need to set `nixpkgs.config.allowUnfreePredicate` in my config, but the value it is set to doesn't seem to work anymore.

**Comment by gytis-ivaskevicius** [comment]
When using flake-utils-plus cases like these can be specified in two areas:
1. For channel-specific configuration (i believe this is what you are looking for) https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/examples/fully-featured-flake.nix#L46-L48
2. Or globally (applied to all channels) https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/examples/fully-featured-flake.nix#L63-L66

**Comment by chvp** [comment]
Yes, I found that, but I build up a list of allowed packages in the nixos modules for my machines, which I don't have access to at that point.

**Comment by gytis-ivaskevicius** [comment]
Interesting use case, admittedly I don't really see much point in doing something like that but ill look into it regardless. I obviously have not tested it but fix should be as simple as this:
Replace this code block: https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/systemFlake.nix#L75-L85
with:
```
```
 {config, ...}:  {
          networking.hostName = name;

          nixpkgs = {
            pkgs = selectedNixpkgs;
            config = selectedNixpkgs.config // config.nixpkgs.config;
          };

          system.configurationRevision = mkIf (self ? rev) self.rev;
          nix.package = mkDefault selectedNixpkgs.nixUnstable;
        }
        ```
(two lines changed, will test it out once I am available)

**Comment by chvp** [comment]
That doesn't seem to work. I think the problem is caused by the `pkgs = selectedNixpkgs;` line. The config applied in `nixpkgs.config` doesn't seem to apply to it, but only to the `nixpkgs` that is used for `nixpkgs.lib.nixosSystem`. [When I remove](https://github.com/chvp/flake-utils-plus/commit/97f6aa2fb92c699057d7199bcf629ad4bc8f4717) the `pkgs = selectedNixpkgs` line (and manually add the overlays I use) it works as expected.

**Comment by gytis-ivaskevicius** [comment]
As of commit states... I Will fix the breaking stuff later, but I am quite happy with the new implementation/dataflow of it

**Comment by gytis-ivaskevicius** [comment]
Pushed some changes, fixed channels switching, few other changes. 
What's unfortunate is that it seems that I will not be able to implement `nixpkgs.config.xyz` with the addition of channel patching (User will be forced to pick one of these features), but we will see. I Will give it some more thought overnight

The next release is planned for the weekend.

**Comment by chvp** [comment]
Ah, that is unfortunate, the easy patching was one of my main reasons for starting to use this. Am I correct in thinking that [this](https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/systemFlake.nix#L63) is the main blocker for the combination? Do you know what the underlying issue is?

**Comment by gytis-ivaskevicius** [comment]
Yes, @chvp, you are correct. More specifically because of this https://github.com/NixOS/nix/issues/4669.

But here is a solution to your specific problem:
TL;DR: implement explicit unfree package definitions in different manner
How?
in your flake.nix
```nix
inputs.nixpkgs.input = nixpkgs;
inputs.nixpkgs.overlayBuilder = channels: [
    (final: prev: { unfree = channels.unfree; })
];

inputs.unfree.input = nixpkgs;
inputs.unfree.config.allowUnfree = true;
```
(untested, should more or less work, might have a typo or two)

And instead of building some fancy array of allowed packages now, you can define unfree packages by typing `pkgs.unfree.google-chrome`

This achieves the same result while being much simpler, easier to implement, and resulting in fewer lines of code

**Comment by chvp** [comment]
Hmm, that is an interesting alternative for making the unfree packages explicit. I'll try that out tomorrow. Thanks for the issue on nix, I subscribed to it to keep updated.

**Comment by chvp** [comment]
It took a while before I was able to get to it, but your solution doesn't seem to work. It refuses to evaluate because of unfree packages.

**Comment by gytis-ivaskevicius** [comment]
I got a strong feeling that this is going to be one of the limitations of Nix (patching + dynamic configuration). Will need to look into it more :/
